core/src/impl/Kokkos_Core.cpp

```diff
@@ -751,6 +775,19 @@ void initialize(int& narg, char* arg[]) {
       else
         skip_device = env_skip_device;
     }
+    if (rdevices > 0) {
+      if (skip_device > 0 && rdevices == 1)
+        Impl::throw_runtime_exception(
+            "Error: cannot KOKKOS_SKIP_DEVICE the only "
+            "KOKKOS_RAND_DEVICE.Raised by Kokkos::initialize(int narg, char* "
+            "argc[]).");
```

Comment:
Formatting looks weird.  Missing a space after the dot at the end go the 1st sentence.